src/renderer/npm-search.tsx

```diff
@@ -0,0 +1,47 @@
+import { SearchResponse } from '@algolia/client-search';
+import algoliasearch, { SearchIndex } from 'algoliasearch/lite';
+
+// See full schema: https://github.com/algolia/npm-search#schema
+interface AlgoliaHit {
+  name: string;
+  version: string;
+  versions: Record<string, string>;
+  _highlightResult: {
+    name: {
+      value: string;
+    };
+  };
+}
+
+class NPMSearch {
+  private index: SearchIndex;
+  private searchCache: Map<string, SearchResponse<AlgoliaHit>>;
+  constructor() {
+    const client = algoliasearch(
+      'OFCNCOG2CU',
+      '4efa2042cf4dba11be6e96e5c394e1a4',
```

Comment:
I am concerned that this could be abused by others, is it possible to confirm that this key has very low permissions?
> BTW. is this managed by the electron account?

Reply:
> is it possible to confirm that this key has very low permissions?

From my discussions with Algolia support, this is a read-only API key that should be safe to put into our client.

> is this managed by the electron account?

The index itself is managed by Algolia: https://github.com/algolia/npm-search

Reply:
Okey